nix/flake.nix

```diff
@@ -0,0 +1,148 @@
+{
+  description = "A GPU-accelerated cross-platform terminal emulator and multiplexer written by @wez and implemented in Rust";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+
+    freetype2 = {
+      url = "github:wez/freetype2/de8b92dd7ec634e9e2b25ef534c54a3537555c11";
+      flake = false;
+    };
+
+    harfbuzz = {
+      url = "github:harfbuzz/harfbuzz/60841e26187576bff477c1a09ee2ffe544844abc";
+      flake = false;
+    };
+
+    libpng = {
+      url = "github:glennrp/libpng/8439534daa1d3a5705ba92e653eda9251246dd61";
+      flake = false;
+    };
+
+    zlib = {
+      url = "github:madler/zlib/cacf7f1d4e3d44d871b605da3b647f07d718623f";
+      flake = false;
+    };
+  };
+
+  outputs = {
+    self,
+    flake-utils,
+    rust-overlay,
+    nixpkgs,
+    freetype2,
+    harfbuzz,
+    libpng,
+    zlib,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      overlays = [(import rust-overlay)];
+      pkgs = import nixpkgs {inherit system overlays;};
+
+      inherit (nixpkgs) lib;
+      inherit (pkgs) stdenv;
+
+      nativeBuildInputs = with pkgs;
+        [
+          installShellFiles
+          ncurses # tic for terminfo
+          pkg-config
+          python3
+
+          rust-bin.stable.latest.minimal
+          rust-bin.stable.latest.clippy
+
+          rust-bin.nightly.latest.rustfmt
+          rust-bin.nightly.latest.rust-analyzer
+        ]
+        ++ lib.optional stdenv.isDarwin perl;
+
+      buildInputs = with pkgs;
+        [
+          fontconfig
+          pkgs.zlib
+        ]
+        ++ lib.optionals stdenv.isLinux [
+          xorg.libX11
+          xorg.libxcb
+          libxkbcommon
+          openssl
+          wayland
+          xorg.xcbutil
+          xorg.xcbutilimage
+          xorg.xcbutilkeysyms
+          xorg.xcbutilwm # contains xcb-ewmh among others
+        ]
+        ++ lib.optionals stdenv.isDarwin [
+          Cocoa
+          CoreGraphics
+          Foundation
+          libiconv
+          UserNotifications
+        ];
+
+      libPath = lib.makeLibraryPath (with pkgs; [libGL vulkan-loader]);
+
+      rustPlatform = pkgs.makeRustPlatform {
+        cargo = pkgs.rust-bin.stable.latest.minimal;
+        rustc = pkgs.rust-bin.stable.latest.minimal;
+      };
+    in {
+      packages.default = rustPlatform.buildRustPackage rec {
+        inherit buildInputs nativeBuildInputs;
```

Comment:
I think you're better of inlining these and using `inputsFrom = self.packages.${system}.default` in the devShell.